lib/types.js

```diff
@@ -407,37 +407,67 @@ module.exports = function (expect) {
                 keys[i] = i;
             }
             if (!this.numericalPropertiesOnly) {
-                Object.keys(obj).forEach(function (key) {
-                    if (!utils.numericalRegExp.test(key)) {
-                        keys.push(key);
-                    }
-                });
+                keys = keys.concat(this.getKeysNonNumerical(obj));
```

Comment:
`keys.push(...this.getKeysNonNumerical(obj))` might be faster, as it won't have to create a new array?